File: result/translate/chatgpt/java2cpp/c++/MAXIMUM_VALUE_CHOICE_EITHER_DIVIDING_CONSIDERING.cpp
int f_gold(int n) {
    int* res = new int[n + 1];
    res[0] = 0;
    res[1] = 1;
    for(int i = 2; i <= n; i++) {
        res[i] = std::max(i, (res[i / 2] + res[i / 3] + res [i / 4] + res[i / 5]));
    }
    return res[n];
}

File: result/translate/chatgpt/python2java/java/MAXIMUM_VALUE_CHOICE_EITHER_DIVIDING_CONSIDERING.java
public List<Integer> f_gold(int n){
    List<Integer> res = new ArrayList<>();
    res.add(0);
    res.add(1);
    int i = 2;
    while(i < n + 1){
        res.add(Math.max(i,(res.get((int)i/2) + res.get((int)i/3) + res.get((int)i/4) + res.get((int)i/5))));
        i++;
    }
    return res.get(n);
}